os/fs/smartfs/smartfs.h

```diff
@@ -254,10 +256,16 @@
  */
 
 struct smartfs_entry_s {
-	uint16_t firstsector;		/* Sector number of the name */
-	uint16_t dsector;			/* Sector number of the directory entry */
+	uint16_t firstsector;			/* First data sector number of the file */
+	uint16_t dsector;			/* Sector number of the directory entry
+						   We will always scan starting from this sector to find space for the new entry.
+						   After finddirentry(), this holds the last directory in the path for the new entry, i.e. immediate parent.
+						   After find_availableentry(), this holds the new entry sector or the last active sector
+						   to which new sector will be chained after writing entry.
+						 */
```

Comment:
For multiline comments, we generally put a star (*) at the start of each line.

Reply:
Will make the change, thanks.